usaspending_api/search/v2/views/spending_over_time.py

```diff
@@ -34,96 +28,78 @@ class SpendingOverTimeVisualizationViewSet(APIView):
 
-        if subawards:
-            queryset = subaward_filter(filters)
+    def database_data_layer(self):
+        if self.subawards:
+            queryset = subaward_filter(self.filters)
+            obligation_column = "amount"
         else:
-            queryset = spending_over_time(filters).values('action_date', 'generated_pragmatic_obligation')
+            queryset = spending_over_time(self.filters)
+            obligation_column = "generated_pragmatic_obligation"
+
+        values = ["fy"]
+        if self.groupings[self.group] == "month":
+            queryset = queryset.annotate(month=FiscalMonth("action_date"), fy=FiscalYear("action_date"))
+            values.append("month")
+
+        elif self.groupings[self.group] == "quarter":
+            queryset = queryset.annotate(quarter=FiscalQuarter("action_date"), fy=FiscalYear("action_date"))
+            values.append("quarter")
+
+        elif self.groupings[self.group] == "fiscal_year":
+            queryset = queryset.annotate(fy=FiscalYear("action_date"))
+
+        queryset = (
+            queryset.values(*values)
+            .annotate(aggregated_amount=Sum(obligation_column))
+            .order_by(*["{}".format(value) for value in values])
+        )
+
+        return queryset, values
+
+    @cache_response()
+    def post(self, request):
+        json_request = self.validate_request_data(request.data)
+        self.group = json_request["group"]
+        self.subawards = json_request["subawards"]
+        self.filters = json_request["filters"]
+
+        db_results, values = self.database_data_layer()
 
         # build response
-        response = {'group': group, 'results': []}
-        nested_order = ''
-
-        # list of time_period objects ie {"fy": "2017", "quarter": "3"} : 1000
-        group_results = OrderedDict()
-
-        # for Subawards we extract data from action_date
-        if subawards:
-            data_set = queryset \
-                .values('award_type') \
-                .annotate(month=ExtractMonth('action_date'), transaction_amount=Sum('amount')) \
-                .values('month', 'fiscal_year', 'transaction_amount')
-        else:
-            # for Awards we Sum generated_pragmatic_obligation for transaction_amount
-            queryset = queryset.values('fiscal_year')
-            if group in ('fy', 'fiscal_year'):
-                data_set = queryset \
-                    .annotate(transaction_amount=Sum('generated_pragmatic_obligation')) \
-                    .values('fiscal_year', 'transaction_amount')
-            else:
-                # quarterly also takes months and aggregates the data
-                data_set = queryset \
-                    .annotate(
-                        month=ExtractMonth('action_date'),
-                        transaction_amount=Sum('generated_pragmatic_obligation')) \
-                    .values('fiscal_year', 'month', 'transaction_amount')
-
-        for record in data_set:
```

Comment:
By using the custom Postgres Extracts defined in `usaspending_api/common/helpers/sql_helpers.py` , the SQL statement returns the data in the final response format. The code no longer needs to manipulate/combine the db data rows into the expected response format